src/traversals/trees_and_forests.jl

```diff
@@ -0,0 +1,43 @@
+default_root_vertex(g) = last(findmax(eccentricities(g)))
+
+function spanning_tree(
+  g::AbstractNamedGraph; root_vertex=default_root_vertex(g), alg::String="BFS"
+)
+  @assert !NamedGraphs.is_directed(g)
+  if alg == "BFS"
+    return undirected_graph(bfs_tree(g, root_vertex))
+  elseif alg == "RandomBFS"
+    return undirected_graph(random_bfs_tree(g, root_vertex))
+  elseif alg == "DFS"
+    return undirected_graph(dfs_tree(g, root_vertex))
+  else
+    error("Algorithm not current supported")
+  end
+end
+
+#Given a graph, split it into its connected components, construct a spanning tree over each of them
+# and take the union.
+function spanning_forest(
+  g::AbstractNamedGraph; spanning_tree_function=g -> spanning_tree(g)
```

Comment:
```suggestion
  g::AbstractNamedGraph; spanning_tree=spanning_tree
```
i.e. shorten the kwarg from `spanning_tree_function` to `spanning_tree`, and also you can just use `f` instead of `x -> f(x)` in general.